src/reader.rs

```diff
@@ -2497,16 +2508,26 @@ impl<'a> Reader<'a> {
                     self.save_screen_state();
                 }
                 ImplicitEvent::PrimaryDeviceAttribute => {
-                    let Some(wait) = &self.blocking_wait else {
-                        // Rogue reply.
-                        return ControlFlow::Continue(());
-                    };
-                    let BlockingWait::Startup(stage) = wait else {
-                        // Rogue reply.
-                        return ControlFlow::Continue(());
+                    // We *received* a response, so we should handle it even if there was a timeout
+                    let stage = match self.blocking_wait() {
+                        Ok(Some(BlockReason::Startup(stage))) => stage.clone(),
+                        Err(BlockingWaitExpired(BlockReason::Startup(stage))) => {
+                            let stage = stage.clone();
+                            self.blocking_wait.reset();
+                            stage
+                        }
+                        Ok(_) => {
+                            // Rogue reply: ignore.
+                            return ControlFlow::Continue(());
+                        }
+                        Err(_) => {
+                            // Unexpected timeout: ignore. Safe to reset at this point
+                            self.blocking_wait.reset();
+                            return ControlFlow::Continue(());
+                        }
                     };
                     match stage {
-                        Queried::NotYet => panic!(),
+                        Queried::NotYet => return ControlFlow::Continue(()),
```

Comment:
this code is very bad but I think this panic was correct.
`Queried::NotYet` is only set initially, and changed before we read anything from stdin that could reach here.
We should get rid of the `Queried::NotYet` enum variant and use a separate static bool instead, to reduce scope.